include/jwt-cpp/jwt.h

```diff
@@ -916,6 +921,110 @@ namespace jwt {
 			const size_t signature_length;
 		};
 
+#ifndef OPENSSL110
+		/**
+		 * \brief Base class for EdDSA family of algorithms
+		 */
+		struct eddsa {
+			/**
+			 * Construct new eddsa algorithm
+			 * \param public_key EdDSA public key in PEM format
+			 * \param private_key EdDSA private key or empty string if not available. If empty, signing will always fail.
+			 * \param public_key_password Password to decrypt public key pem.
+			 * \param private_key_password Password to decrypt private key pem.
+			 * \param md Pointer to hash function
+			 * \param name Name of the algorithm
+			 */
+			eddsa(const std::string& public_key, const std::string& private_key, const std::string& public_key_password, const std::string& private_key_password, const EVP_MD* (*md)(), std::string  name)
+				: md(md), alg_name(std::move(name))
+			{
+				if (!private_key.empty()) {
+					pkey = helper::load_private_key_from_string(private_key, private_key_password);
+				}
+				else if (!public_key.empty()) {
+					pkey = helper::load_public_key_from_string(public_key, public_key_password);
+				}
+				else
+					throw ecdsa_exception(error::ecdsa_error::load_key_bio_read);
+			}
+			/**
+			 * Sign jwt data
+			 * \param data The data to sign
+			 * \param ec error_code filled with details on error
+			 * \return EdDSA signature for the given data
+			 */
+			std::string sign(const std::string& data, std::error_code& ec) const {
+				ec.clear();
+				std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_free)> ctx(EVP_MD_CTX_create(), EVP_MD_CTX_free);
+				if (!ctx) {
+					ec = error::signature_generation_error::create_context_failed;
+					return {};
+				}
+				if (!EVP_DigestSignInit(ctx.get(), nullptr, nullptr, nullptr, pkey.get())) {
+					ec = error::signature_generation_error::signinit_failed;
+					return {};
+				}
+
+				std::string res;
+				size_t len = EVP_PKEY_size(pkey.get());
+				res.resize(len);
+
+				if (EVP_DigestSign(ctx.get(),
+						reinterpret_cast<unsigned char*>(&res[0]), &len,
+						reinterpret_cast<const unsigned char*>(data.data()), data.size()) != 1) {
+					ec = error::signature_generation_error::signfinal_failed;
+					return {};
+				}
+
+				res.resize(len);
+				return res;
+			}
+
+			/**
+			 * Check if signature is valid
+			 * \param data The data to check signature against
+			 * \param signature Signature provided by the jwt
+			 * \param ec Filled with details on error
+			 */
+			void verify(const std::string& data, const std::string& signature, std::error_code& ec) const {
+				ec.clear();
+#ifdef OPENSSL10
+				std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_destroy)> ctx(EVP_MD_CTX_create(), EVP_MD_CTX_destroy);
+#else
+				std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_free)> ctx(EVP_MD_CTX_create(), EVP_MD_CTX_free);
+#endif
```

Comment:
Since OpenSSL 1.1.1 or greater is required!
```suggestion
				std::unique_ptr<EVP_MD_CTX, decltype(&EVP_MD_CTX_free)> ctx(EVP_MD_CTX_create(), EVP_MD_CTX_free);
```

Reply:
Do we really wanna drop support for everything less then 1.1.1 ?
I think removing support for eddsa (assuming its not possible to implement without 1.1.1) in lower versions would be a better solution.

Reply:
thats what's currently proposed, this is just a copy paste error

Reply:
Whoops, missed that one. I did remove the other such place. :)

And yes, the library builds on 1.1.0 without EdDSA right now; and builds with 1.1.1 with EdDSA support.